src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -250,7 +250,7 @@ private async Task<string> GetLatestMatchingBundleVersionAsync(HttpClient httpCl
             return matchingBundleVersion;
         }
 
-        internal static string FindBestVersionMatch(VersionRange versionRange, IEnumerable<string> versions, string bundleId, FunctionsHostingConfigOptions configOption)
+        internal string FindBestVersionMatch(VersionRange versionRange, IEnumerable<string> versions, string bundleId, FunctionsHostingConfigOptions configOption)
         {
             var bundleVersions = versions.Select(p =>
```

Comment:
I think we are going to be enumerating this list multiple times with these added checks. We should realize this to a list and then pass that around instead of `IEnumerable`. Make sure to include the `OrderByDescending` before calling `.ToList()`.

Also can this entire collection LINQ operations can be moved into `ResolvePlatformReleaseChannelVersion`?